File: src/main/java/co/com/fsighttech/imp/issuing/bo/configuration/MasterVariableData.java
/**
 * 
 */
package co.com.fsighttech.imp.issuing.bo.configuration;

import lombok.Getter;
import lombok.Setter;

/**
 * this class represents a master variable data 
 * 
 * @author jtorres
 *
 */
@Getter
@Setter
public class MasterVariableData {
	private Long   id;
	private Integer sequence;
	private String code;
	private String description;
	private String abbreviation;
	private Long parentCode;
	private InputProperties inputProperties;
	
}

File: src/main/resources/static/src/app/agent-portal/shared/model/configuration/master-variable-data.ts
import { JsonMember, JsonObject } from 'typedjson-npm/src/typed-json';
import {SelectItem} from 'primeng/primeng';

import { InputProperties } from './input-properties';
import { VariableDataOption } from './variable-data-option';
import { Field } from '../../../../shared/dyna-field/field';



/**
* this object represents a master variable data for policy
*
* @author ccasallas
*/
@JsonObject
export class MasterVariableData extends Field {
    @JsonMember
    id: number;
    @JsonMember
    sequence: number;
    @JsonMember
    code: string;
    @JsonMember
    description: string;
    @JsonMember
    abbreviation: string;
    @JsonMember
    parentCode: number;
    @JsonMember
    inputProperties : InputProperties;

    getFieldType(): string {
        return this.inputProperties.typeVariable;
    }
    
    initPlaceholder(): void {
        this.placeholder = this.inputProperties.label;
    }
    
    initSelectItems(): void {
        this.selectItems = [];
        //additional options
        this.inputProperties.optionList.forEach((option: VariableDataOption ) => {
           this.selectItems.push({label:  option.description, value : option});
        });
        
    }
}